services/src/main/java/org/keycloak/authentication/authenticators/x509/CertificateValidator.java

```diff
@@ -468,6 +469,24 @@ public CertificateValidator validateExtendedKeyUsage() throws GeneralSecurityExc
         return this;
     }
 
+    public CertificateValidator validateTimestamps(boolean isValidationEnabled) throws GeneralSecurityException {
+        if (!isValidationEnabled) {
+            return this;
+        }
+        X509Certificate x509Certificate = _certChain[_certChain.length - 1];
+        // this might create time-offset issues but those will probably be negligible for certificates should
+        // be renewed and exchanged in due time
+        if (x509Certificate.getNotBefore().getTime() > Instant.now().toEpochMilli()) {
```

Comment:
Should use org.keycloak.common.util.Time as that supports tweaking the time for testing purposes

Reply:
done